doc/building.md

```diff
@@ -115,8 +125,10 @@ You can install both compilers and select the one to be used by `./configure`.
 
  - (obviously) clone this repo or acquire the sources some other way
  - make sure you have everything from the [dependency list](#dependencies)
- - `./configure --compiler=clang` will prepare building
+ - For *nix: `./configure --compiler=clang` will prepare building
   - You could also use `./configure --mode=release --compiler=gcc` here
+ - For Mac OS X: use instead:  `./configure --cpp-compiler=clang++-3.5`
+  -  Or: `./configure --mode=release --compiler=g++-4.9`
```

Comment:
Brew  installs these compilers with the version number as a part of the executable name, so that they do not not interfere with already-installed Apple tools. Brew doesn't create symlinks without the version number (unlike Linux). 

If someone uses `--compiler=clang` then it would run the Apple-installed clang and fail. The same would happen with gcc.